venona/cmd/start.go

```diff
@@ -133,9 +143,54 @@ func run(options startOptions) {
 	}
 	dieOnError(agent.Start())
 
-	server := server.Server{
+	serverMode := server.Release
+	if options.verbose {
+		serverMode = server.Debug
+	}
+	server, err := server.New(&server.Options{
 		Port:   fmt.Sprintf(":%s", options.serverPort),
 		Logger: log.New("module", "server"),
-	}
+		Mode:   serverMode,
+	})
+	dieOnError(err)
+
+	go handleSignals(context.Background(), &agent, &server, log)
+
 	dieOnError(server.Start())
 }
+
+func handleSignals(ctx context.Context, a *agent.Agent, s *server.Server, log logger.Logger) {
+	sigChan := make(chan os.Signal, 10)
+	receivedTerminationReq := false
+	receivedTerminationReqMux := sync.Mutex{}
+
+	handleSignal(sigChan, syscall.SIGTERM, syscall.SIGINT) // sent by k8s
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case sig := <-sigChan:
+			switch sig {
+			case syscall.SIGTERM, syscall.SIGINT:
+				go func() {
```

Comment:
Not sure why we need run another routine here

Reply:
running another routine to allow the "force-stop" option, don't want to block the handling of signals and wait for server to finish.